core/src/main/java/com/klaytn/caver/transaction/AbstractFeeDelegatedTransaction.java

```diff
@@ -0,0 +1,333 @@
+        SingleKeyring keyring = KeyringFactory.createFromPrivateKey(keyString);
+
+        return signAsFeePayer(keyring, hasher);
+    }
+
+    /**
+     * Signs using all private keys used in the role defined in the Keyring instance as a fee payer.
+     * It sets index and Hasher default value.
+     *    - signer : TransactionHasher.getHashForSignature()
+     * @param keyring The Keyring instance.
+     * @return AbstractFeeDelegatedTransaction
+     * @throws IOException
+     */
+    public AbstractFeeDelegatedTransaction signAsFeePayer(AbstractKeyring keyring) throws IOException {
+        return signAsFeePayer(keyring, TransactionHasher::getHashForFeePayerSignature);
+    }
+
+    /**
+     * Signs using all private keys used in the role defined in the keyring instance as a fee payer.
+     * @param keyring The Keyring instance.
+     * @param index The index of private key to use in Keyring instance.
+     * @return AbstractFeeDelegatedTransaction
+     * @throws IOException
+     */
+    public AbstractFeeDelegatedTransaction signAsFeePayer(AbstractKeyring keyring, int index) throws IOException {
+        return signAsFeePayer(keyring, index, TransactionHasher::getHashForFeePayerSignature);
+    }
+
+    /**
+     * Signs using all private keys used in the role defined in the Keyring instance as a fee payer.
+     * @param keyring The Keyring instance.
+     * @param hasher The function to get hash of transaction.
+     * @return AbstractFeeDelegatedTransaction
+     * @throws IOException
+     */
+    public AbstractFeeDelegatedTransaction signAsFeePayer(AbstractKeyring keyring, Function<AbstractFeeDelegatedTransaction, String> hasher) throws IOException {
+        if(!this.getFeePayer().toLowerCase().equals(keyring.getAddress().toLowerCase())) {
+            throw new IllegalArgumentException("The feePayer address of the transaction is different with the address of the keyring to use.");
+        }
+
+        this.fillTransaction();
+        int role = AccountKeyRoleBased.RoleGroup.FEE_PAYER.getIndex();
+
+        String hash = hasher.apply(this);
+        List<SignatureData> sigList = keyring.sign(hash, Numeric.toBigInt(this.getChainId()).intValue(), role);
+
+        this.appendFeePayerSignatures(sigList);
+
+        return this;
+    }
+
+    /**
+     * Signs to the transaction with a private key in the Keyring instance as a fee payer.
+     * @param keyring The Keyring instance.
+     * @param index The index of private key to use in Keyring instance.
+     * @param hasher The function to get hash of transaction.
+     * @return AbstractFeeDelegatedTransaction
+     * @throws IOException
+     */
+    public AbstractFeeDelegatedTransaction signAsFeePayer(AbstractKeyring keyring, int index, Function<AbstractFeeDelegatedTransaction, String> hasher) throws IOException {
+        if(!this.getFeePayer().toLowerCase().equals(keyring.getAddress().toLowerCase())) {
```

Comment:
How about set fee payer's address with keyring.address if this.getFeepayer() == "0x"?

Reply:
Okay, I added this suggestion.